xonsh/environ.py

```diff
@@ -82,21 +92,15 @@ def format_prompt(template=default_prompt):
     + NO_COLOR -- Resets any previously used color codes
     """
     env = builtins.__xonsh_env__
-    cwd = env['PWD']
-    branch = current_branch(cwd=cwd)
-    branch = '' if branch is None else ' ' + branch
-    p = template.format(
-            user=env.get('USER', '<user>'),
-            hostname=socket.gethostname(),
-            cwd=cwd.replace(env['HOME'], '~'),
-            curr_branch=branch,
-            **TERM_COLORS
-            )
-    return p
+    fmt = env.get('FORMAT_DICT', FORMAT_DICT)
+    fmt = {k:v for (k,v) in fmt.items() if '{{{}}}'.format(k) in template}
```

Comment:
I think that this might deserve a regular expression, because what if someone wanted to do something with more advanced string formatting such as `'{test:>10}'`. I think the right regex is something like:

```python
'[{]([^:]*):?.*?[}]'
```